### ISSUE TITLE
Question about Training Details for Stage 1 SAM Masks

### ISSUE THREAD

**Issue by wushixuan238** [andrehuang/loftup#9]
Dear Authors,

Thank you for your excellent and outstanding work!

I am studying your paper carefully and have a few questions regarding the training details in Stage 1, specifically about the construction of the pseudo-ground truth FMask−Bicubic using SAM masks:

1. Regarding the formation process of the final FMask−Bicubic[m] (features refined within mask m) is computed.

 I would like to understand how the final FMask−Bicubic map, which covers the entire image and is used for loss calculation, is obtained by aggregating the results from processing all individual masks m i∈M? Specifically, if different SAM masks overlap, how are the final feature values determined for pixels in these overlapping regions? For instance, are they overwritten by the values from the last processed mask that includes the pixel, or is there another fusion strategy involved?

2. Regarding the number of masks (N) used per image:
When processing each image, is the number of SAM masks N a fixed value?
I noted that some public information about the SAM/SA1B dataset mentions an "Average masks per image: 100". In your implementation, do you process approximately 100 masks for every image, or do you use all N masks that are actually available for that specific image in the SA1B dataset (where N can vary per image)?

Looking forward to your reply! Thank you again!

**Comment by andrehuang** [comment]
Hi,

Thanks for your interest in our work!

We will release our training code soon in late May or early June. Please stay tuned! Here are the answers to your questions:

1. Regarding the formation process of the final FMask−Bicubic[m] (features refined within mask m) is computed.

To goal of using SAM masks to adjust the features is to make boundaries sharper, and to not overfit to any semantics. Therefore, we would like to keep the finest details. Specifically, we apply the mask adjustment from large to small masks, i.e., each pixel only takes values from the smallest mask that contains it.

2. Regarding the number of masks (N) used per image.

For data parallelization, we set N to be 150 for all images (and add empty masks if the actual number is smaller than 150). As you noticed, the avg number of masks is around 100, so 150 is a quite safe choice and won't waste too much memory.

Hope our answers help!

Best,
Haiwen

**Comment by wushixuan238** [comment]
Thank you for your response. I appreciate you taking the time to reply. I understand now. Thanks again!
